### PR TITLE
Setup pre-commit hooks

### PR DESCRIPTION
**Pull request by remoteeng00**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -47,53 +46,50 @@ def read(fname):
...
 ]
 
 # extra requirements to run biothings.hub
 hub_requires = [
-    'beautifulsoup4',   # used in dumper.GoogleDriveDumper
-    'aiocron==1.8',     # setup scheduled jobs
-    'aiohttp==3.8.1',   # elasticsearch requires aiohttp>=3,<4
+    'beautifulsoup4',  # used in dumper.GoogleDriveDumper
+    'aiocron==1.8',  # setup scheduled jobs
+    'aiohttp==3.8.1',  # elasticsearch requires aiohttp>=3,<4
```
Comment (newgene) [inline setup.py:63]: This is a case I'm kind of like the the old style better. Wonder if have a way to control this behavior.

But also it's not a big deal of course.

File: setup.py [hunk 2]
```diff
@@ -133,6 +129,6 @@ def read(fname):
     extras_require={
         'web_extra': web_extra_requires,
         'hub': hub_requires,
-        'dev': web_extra_requires + hub_requires + dev_requires + docs_requires
+        'dev': web_extra_requires + hub_requires + dev_requires + docs_requires,
```
Comment (newgene) [inline setup.py:132]: is adding this trailing comma a preferred way now? Remember it was recommended to remove it, maybe this rule changed now.

File: setup.py [hunk 3]
```diff
@@ -47,53 +46,50 @@ def read(fname):
     'singledispatchmethod; python_version < "3.8.0"',
     'dataclasses; python_version < "3.7.0"',
     'PyYAML>=5.1',
-    'orjson>=3.6.1',    # this is a faster json lib support inf/nan and datetime
-                        # v3.6.1 is the last version supports Python 3.6
+    'orjson>=3.6.1',  # this is a faster json lib support inf/nan and datetime
+    # v3.6.1 is the last version supports Python 3.6
```
Comment (newgene) [inline setup.py:50]: I also don't like this line of fix. The line of comment was part of the line above.

File: setup.py [hunk 4]
```diff
@@ -47,53 +46,50 @@ def read(fname):
...
+    'requests-aws4auth',  # aws s3 auth requests for autohub
+    'networkx>=2.1,<2.6',  # datatransform
+    'biothings_client>=0.2.6',  # datatransform (api client)
 ]
 
 # extra requirements to develop biothings
 dev_requires = [
     'pytest',
     'pytest-mock',
+    'pre-commit==2.17.0',
```
Comment (newgene) [inline setup.py:88]: is `fourmat` also required here now?

Comment (remoteeng00) [reply setup.py:88]: no it isn't, we define it in the [.pre-commit-config.yaml](https://github.com/newgene/biothings.api/pull/53/files/0bfc9a2cb6451eb269d2aabafd796d2744ecd8b6#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R12)

**Pull request merged**